### PR TITLE
Added modifiable global default for tensor string formatting.

### PR DESCRIPTION
**Pull request by NiklasGustafsson**
This PR allows you to set the default tensor string format using `torch.TensorStringStyle = ...`

### CODE REVIEW

**Comment by NiklasGustafsson** [comment]
@George2019, @dayo05 -- please take a look at this PR. The idea is to be able to set the global default string format rather than have the default be a default argument. 

This allows you to control the behavior of all `ToString()`, `print()`, and `str()` that don't take a style parameter. 'Julia' is still the static default, so you have to set `torch.TensorStringStyle = TensorStringStyle.NumPy;` to change it. Once you do that, though, everything will obey it.

File: src/TorchSharp/Tensor/Tensor.cs [hunk 1]
```diff
@@ -6679,6 +6679,7 @@ public Tensor where(Tensor condition, Tensor y)
                     return ToMetadataString();
```
Comment (kaiidams) [inline src/TorchSharp/Tensor/Tensor.cs:6679]: Doc comment needs update. ` /// The style to use -- either 'metadata,' 'julia,' or 'numpy'`

**Review by kaiidams — DISMISSED** [review]

**Comment by dayo05** [comment]
LGTM!

**Comment by GeorgeS2019** [comment]
@NiklasGustafsson  I could not get torch.metadata to work!

**Comment by GeorgeS2019** [comment]
@dayo05  what u mean? 
>LGTM!

**Comment by NiklasGustafsson** [comment]
> @NiklasGustafsson I could not get torch.metadata to work!

Try TensorStringStyle.Metadata

**Comment by NiklasGustafsson** [comment]
> LGTM!



> @dayo05 what u mean?
> 
> > LGTM!

LGTM == Looks Good To Me

**Comment by GeorgeS2019** [comment]
@NiklasGustafsson 

Why on notebook, whenever I use e.g. .`str() ` I get this as additional message?

> warning CS1701: Assuming assembly reference 'System.Runtime.Numerics, Version=4.1.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' used by 'TorchSharp' matches identity 'System.Runtime.Numerics, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' of 'System.Runtime.Numerics', you may need to supply runtime policy

**Comment by tarekgh** [comment]
@GeorgeS2019 do you have link to that notebook? Maybe the notebook is using .NET Framework and referencing the old runtime version of System.Numerics.

**Comment by GeorgeS2019** [comment]
FYI: https://github.com/dotnet/interactive/issues/762

**Comment by NiklasGustafsson** [comment]
@GeorgeS2019 -- there must be something funky in your setup for that to happen. I've never seen anything like that going on with any notebook. Is it one of the books from TorchSharpExamples?

**Comment by GeorgeS2019** [comment]
I run out of idea :-(

I compare using local and remote nuget TorchSharp. same happens for the TorchSharpExamples notebooks. Not sure where the problems :-(

**Comment by GeorgeS2019** [comment]
>  referencing the old runtime version of System.Numerics.

I cleaned up my .nuget packages and all the old System.Runtime.Numerics, e.g. Version=4.1.2.0 to ONLY the latest 4.3.0 
No effect.

**Pull request merged**